networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,139 @@
+import networkx as nx
+from networkx import NetworkXError
+from networkx.utils import not_implemented_for
+
+__all__ = [
+    "tree_broadcast_center",
+    "tree_broadcast_time",
+]
+
+
+def _get_max_broadcast_value(G, U, v, values):
+    adj = sorted(set(G.neighbors(v)) & U, key=lambda u: values[u], reverse=True)
+    return max(values[u] + i for i, u in enumerate(adj, start=1))
+
+
+def _get_broadcast_centers(G, v, values, target):
+    adj = sorted(set(G.neighbors(v)), key=lambda u: values[u], reverse=True)
+    j = min(i for i, u in enumerate(adj, start=1) if values[u] + i == target)
+    return set([v] + adj[:j])
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def tree_broadcast_center(G):
+    """Return the Broadcast Center of the tree G.
+
+    This is a linear algorithm for determining the broadcast center of any tree with N vertices,
+    as a by-product it can also determine the broadcast number of any vertex in the tree.
+    The broadcast number of a vertex v in a graph G is the minimum number of time units required to broadcast from v.
+    The broadcast center of a graph G is one of the vertices having minimum broadcast number.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+        The graph should be an undirected tree
+
+    Returns
+    -------
+    BC : (int, set) tuple
+        minimum broadcast number of the tree, set of broadcast centers
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        If the graph is directed or is a multigraph.
+
+    References
+    ----------
+    .. [1] Slater, P.J., Cockayne, E.J., Hedetniemi, S.T,
+       Information dissemination in trees. SIAM J.Comput. 10(4), 692–701 (1981)
+    """
+    # Assert that the graph G is a tree
+    if not nx.is_tree(G):
+        NetworkXError("Your graph is not a tree")
+    # step 0
+    if G.number_of_nodes() == 2:
+        return 1, set(G.nodes())
+    elif G.number_of_nodes() == 1:
+        return 0, set(G.nodes())
+
+    # step 1
+    U = {node for node, deg in G.degree if deg == 1}
+    values = {n: 0 for n in U}
+    T = G.copy()
+    T.remove_nodes_from(U)
+
+    # step 2
+    W = {node for node, deg in T.degree if deg == 1}
+    values.update((w, G.degree[w] - 1) for w in W)
+
+    # step 3
+    while T.number_of_nodes() >= 2:
+        # step 4
+        w = min(W, key=lambda n: values[n])
+        v = next(T.neighbors(w))
+
+        # step 5
+        U.add(w)
+        W.remove(w)
+        T.remove_node(w)
+
+        # step 6
+        if T.degree(v) == 1:
+            # update t(v)
+            values.update({v: _get_max_broadcast_value(G, U, v, values)})
+            W.add(v)
+
+    # step 7
+    v = nx.utils.arbitrary_element(T)
+    b_T = _get_max_broadcast_value(G, U, v, values)
+    return b_T, _get_broadcast_centers(G, v, values, b_T)
```

Comment:
the only change I have made to the original function Is here, I am now returning a tuple of (b_T, b_C).

Reply:
I decided to return both since the algorithm is able to calculate both at the same time (one comes at the byproduct of the other).
I think it could work to add a function just for broadcast-time, but the content would be exactly the same (pretty much, except the return statement). 
What are your thoughts on this? Is there an efficient way to divide this into two functions without too much code duplication, some shortcut perhaps? (because I agree it would be better to separate them).